Watching for file changes with StatReloader
Watching for file changes with StatReloader
C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\portfolio\settings.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\portfolio\settings.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\portfolio\settings.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\portfolio\settings.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\portfolio\settings.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\portfolio\settings.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\portfolio\settings.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Watching for file changes with StatReloader
C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\portfolio\settings.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Watching for file changes with StatReloader
/Users/vitor/Documents/GitHub/VitorBarbosa/projects/models.py changed, reloading.
Watching for file changes with StatReloader
/Users/vitor/Documents/GitHub/VitorBarbosa/projects/admin.py changed, reloading.
Watching for file changes with StatReloader
/Users/vitor/Documents/GitHub/VitorBarbosa/projects/views.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Not Found: /favicon.ico
Starting new HTTPS connection (1): api.cloudinary.com:443
https://api.cloudinary.com:443 "POST /v1_1/dtbtvoewx/image/upload HTTP/1.1" 200 None
https://api.cloudinary.com:443 "POST /v1_1/dtbtvoewx/image/upload HTTP/1.1" 200 None
Internal Server Error: /admin/projects/project/2/change/
Traceback (most recent call last):
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "projects_project_additio_project_id_projectimage__bed02e58_uniq"
DETAIL:  Key (project_id, projectimage_id)=(2, 1) already exists.


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/contrib/admin/options.py", line 688, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/utils/decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/views/decorators/cache.py", line 62, in _wrapper_view_func
    response = view_func(request, *args, **kwargs)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/contrib/admin/sites.py", line 242, in inner
    return view(request, *args, **kwargs)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/contrib/admin/options.py", line 1889, in change_view
    return self.changeform_view(request, object_id, form_url, extra_context)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/utils/decorators.py", line 46, in _wrapper
    return bound_method(*args, **kwargs)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/utils/decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/contrib/admin/options.py", line 1747, in changeform_view
    return self._changeform_view(request, object_id, form_url, extra_context)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/contrib/admin/options.py", line 1799, in _changeform_view
    self.save_related(request, form, formsets, not add)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/contrib/admin/options.py", line 1255, in save_related
    self.save_formset(request, form, formset, change=change)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/contrib/admin/options.py", line 1243, in save_formset
    formset.save()
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/forms/models.py", line 784, in save
    return self.save_existing_objects(commit) + self.save_new_objects(commit)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/forms/models.py", line 944, in save_new_objects
    self.new_objects.append(self.save_new(form, commit=commit))
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/forms/models.py", line 1142, in save_new
    return super().save_new(form, commit=commit)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/forms/models.py", line 757, in save_new
    return form.save(commit=commit)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/forms/models.py", line 542, in save
    self.instance.save()
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/models/base.py", line 814, in save
    self.save_base(
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/models/base.py", line 877, in save_base
    updated = self._save_table(
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/models/base.py", line 1020, in _save_table
    results = self._do_insert(
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/models/base.py", line 1061, in _do_insert
    return manager._insert(
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/models/query.py", line 1805, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/models/sql/compiler.py", line 1822, in execute_sql
    cursor.execute(sql, params)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/backends/utils.py", line 102, in execute
    return super().execute(sql, params)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/backends/utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.IntegrityError: duplicate key value violates unique constraint "projects_project_additio_project_id_projectimage__bed02e58_uniq"
DETAIL:  Key (project_id, projectimage_id)=(2, 1) already exists.

/Users/vitor/Documents/GitHub/VitorBarbosa/projects/models.py changed, reloading.
Watching for file changes with StatReloader
/Users/vitor/Documents/GitHub/VitorBarbosa/projects/views.py changed, reloading.
Watching for file changes with StatReloader
/Users/vitor/Documents/GitHub/VitorBarbosa/projects/admin.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /project/2/
Traceback (most recent call last):
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/projects/views.py", line 26, in project_detail
    return render(request, template, {'project': project})
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/shortcuts.py", line 24, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/loader.py", line 15, in get_template
    return engine.get_template(template_name)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/backends/django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/engine.py", line 175, in get_template
    template, origin = self.find_template(template_name)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/engine.py", line 157, in find_template
    template = loader.get_template(name, skip=skip)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/loaders/cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/loaders/base.py", line 28, in get_template
    return Template(
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 200, in compile_nodelist
    return parser.parse()
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 513, in parse
    raise self.error(token, e)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 511, in parse
    compiled_result = compile_func(self, token)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/loader_tags.py", line 293, in do_extends
    nodelist = parser.parse()
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 513, in parse
    raise self.error(token, e)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 511, in parse
    compiled_result = compile_func(self, token)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/loader_tags.py", line 232, in do_block
    nodelist = parser.parse(("endblock",))
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 518, in parse
    self.unclosed_block_tag(parse_until)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 581, in unclosed_block_tag
    raise self.error(token, msg)
django.template.exceptions.TemplateSyntaxError: Unclosed tag on line 34: 'block'. Looking for one of: endblock.
/Users/vitor/Documents/GitHub/VitorBarbosa/projects/models.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Starting new HTTPS connection (1): api.cloudinary.com:443
https://api.cloudinary.com:443 "POST /v1_1/dtbtvoewx/image/upload HTTP/1.1" 200 None
Not Found: /project/2/projectCarousel2
/Users/vitor/Documents/GitHub/VitorBarbosa/projects/views.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Not Found: /favicon.ico
Not Found: /favicon.ico
/Users/vitor/Documents/GitHub/VitorBarbosa/projects/views.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Not Found: /favicon.ico
Starting new HTTPS connection (1): api.cloudinary.com:443
https://api.cloudinary.com:443 "POST /v1_1/dtbtvoewx/image/upload HTTP/1.1" 200 None
Resetting dropped connection: api.cloudinary.com
https://api.cloudinary.com:443 "POST /v1_1/dtbtvoewx/image/upload HTTP/1.1" 200 None
https://api.cloudinary.com:443 "POST /v1_1/dtbtvoewx/image/upload HTTP/1.1" 200 None
https://api.cloudinary.com:443 "POST /v1_1/dtbtvoewx/image/upload HTTP/1.1" 200 None
https://api.cloudinary.com:443 "POST /v1_1/dtbtvoewx/image/upload HTTP/1.1" 200 None
https://api.cloudinary.com:443 "POST /v1_1/dtbtvoewx/image/upload HTTP/1.1" 200 None
https://api.cloudinary.com:443 "POST /v1_1/dtbtvoewx/image/upload HTTP/1.1" 200 None
https://api.cloudinary.com:443 "POST /v1_1/dtbtvoewx/image/upload HTTP/1.1" 200 None
https://api.cloudinary.com:443 "POST /v1_1/dtbtvoewx/image/upload HTTP/1.1" 200 None
https://api.cloudinary.com:443 "POST /v1_1/dtbtvoewx/image/upload HTTP/1.1" 200 None
https://api.cloudinary.com:443 "POST /v1_1/dtbtvoewx/image/upload HTTP/1.1" 200 None
https://api.cloudinary.com:443 "POST /v1_1/dtbtvoewx/image/upload HTTP/1.1" 200 None
https://api.cloudinary.com:443 "POST /v1_1/dtbtvoewx/image/upload HTTP/1.1" 200 None
https://api.cloudinary.com:443 "POST /v1_1/dtbtvoewx/image/upload HTTP/1.1" 200 None
/Users/vitor/Documents/GitHub/VitorBarbosa/projects/models.py changed, reloading.
Watching for file changes with StatReloader
/Users/vitor/Documents/GitHub/VitorBarbosa/projects/views.py changed, reloading.
Watching for file changes with StatReloader
/Users/vitor/Documents/GitHub/VitorBarbosa/projects/views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /project/3/
Traceback (most recent call last):
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.UndefinedColumn: column projects_project.trailer_url does not exist
LINE 1: ...oject"."featured", "projects_project"."template", "projects_...
                                                             ^


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/projects/views.py", line 24, in project_detail
    project = get_object_or_404(Project, pk=pk)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/shortcuts.py", line 85, in get_object_or_404
    return queryset.get(*args, **kwargs)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/models/query.py", line 633, in get
    num = len(clone)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/models/query.py", line 380, in __len__
    self._fetch_all()
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/models/query.py", line 1881, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/models/query.py", line 91, in __iter__
    results = compiler.execute_sql(
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/models/sql/compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/backends/utils.py", line 102, in execute
    return super().execute(sql, params)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/backends/utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.ProgrammingError: column projects_project.trailer_url does not exist
LINE 1: ...oject"."featured", "projects_project"."template", "projects_...
                                                             ^

Watching for file changes with StatReloader
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Internal Server Error: /project/3/
Traceback (most recent call last):
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/projects/views.py", line 26, in project_detail
    return render(request, template, {'project': project})
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/shortcuts.py", line 24, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/loader.py", line 15, in get_template
    return engine.get_template(template_name)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/backends/django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/engine.py", line 175, in get_template
    template, origin = self.find_template(template_name)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/engine.py", line 157, in find_template
    template = loader.get_template(name, skip=skip)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/loaders/cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/loaders/base.py", line 28, in get_template
    return Template(
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 200, in compile_nodelist
    return parser.parse()
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 513, in parse
    raise self.error(token, e)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 511, in parse
    compiled_result = compile_func(self, token)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/loader_tags.py", line 293, in do_extends
    nodelist = parser.parse()
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 513, in parse
    raise self.error(token, e)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 511, in parse
    compiled_result = compile_func(self, token)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/loader_tags.py", line 232, in do_block
    nodelist = parser.parse(("endblock",))
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 513, in parse
    raise self.error(token, e)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 511, in parse
    compiled_result = compile_func(self, token)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/defaulttags.py", line 953, in do_if
    nodelist = parser.parse(("elif", "else", "endif"))
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 484, in parse
    raise self.error(token, e)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 482, in parse
    filter_expression = self.compile_filter(token.contents)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 600, in compile_filter
    return FilterExpression(token, self)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 698, in __init__
    filter_func = parser.find_filter(filter_name)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 606, in find_filter
    raise TemplateSyntaxError("Invalid filter: '%s'" % filter_name)
django.template.exceptions.TemplateSyntaxError: Invalid filter: 'replace'
Not Found: /favicon.ico
Internal Server Error: /project/3/
Traceback (most recent call last):
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/projects/views.py", line 26, in project_detail
    return render(request, template, {'project': project})
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/shortcuts.py", line 24, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/loader.py", line 15, in get_template
    return engine.get_template(template_name)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/backends/django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/engine.py", line 175, in get_template
    template, origin = self.find_template(template_name)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/engine.py", line 157, in find_template
    template = loader.get_template(name, skip=skip)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/loaders/cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/loaders/base.py", line 28, in get_template
    return Template(
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 200, in compile_nodelist
    return parser.parse()
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 513, in parse
    raise self.error(token, e)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 511, in parse
    compiled_result = compile_func(self, token)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/loader_tags.py", line 293, in do_extends
    nodelist = parser.parse()
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 513, in parse
    raise self.error(token, e)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 511, in parse
    compiled_result = compile_func(self, token)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/loader_tags.py", line 232, in do_block
    nodelist = parser.parse(("endblock",))
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 513, in parse
    raise self.error(token, e)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 511, in parse
    compiled_result = compile_func(self, token)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/defaulttags.py", line 953, in do_if
    nodelist = parser.parse(("elif", "else", "endif"))
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 484, in parse
    raise self.error(token, e)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 482, in parse
    filter_expression = self.compile_filter(token.contents)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 600, in compile_filter
    return FilterExpression(token, self)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 698, in __init__
    filter_func = parser.find_filter(filter_name)
  File "/Users/vitor/Documents/GitHub/VitorBarbosa/myenv/lib/python3.9/site-packages/django/template/base.py", line 606, in find_filter
    raise TemplateSyntaxError("Invalid filter: '%s'" % filter_name)
django.template.exceptions.TemplateSyntaxError: Invalid filter: 'replace'
/Users/vitor/Documents/GitHub/VitorBarbosa/projects/templatetags/custom_filters.py changed, reloading.
Watching for file changes with StatReloader
/Users/vitor/Documents/GitHub/VitorBarbosa/projects/models.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Not Found: /favicon.ico
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Internal Server Error: /project/2/
Traceback (most recent call last):
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\projects\views.py", line 26, in project_detail
    return render(request, template, {'project': project})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\shortcuts.py", line 24, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\engine.py", line 175, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\engine.py", line 157, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 200, in compile_nodelist
    return parser.parse()
           ^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 513, in parse
    raise self.error(token, e)
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 511, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loader_tags.py", line 293, in do_extends
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 513, in parse
    raise self.error(token, e)
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 511, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loader_tags.py", line 232, in do_block
    nodelist = parser.parse(("endblock",))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 513, in parse
    raise self.error(token, e)
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 511, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\defaulttags.py", line 953, in do_if
    nodelist = parser.parse(("elif", "else", "endif"))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 484, in parse
    raise self.error(token, e)
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 482, in parse
    filter_expression = self.compile_filter(token.contents)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 600, in compile_filter
    return FilterExpression(token, self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 698, in __init__
    filter_func = parser.find_filter(filter_name)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 606, in find_filter
    raise TemplateSyntaxError("Invalid filter: '%s'" % filter_name)
django.template.exceptions.TemplateSyntaxError: Invalid filter: 'youtube_embed'
Not Found: /favicon.ico
C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\projects\views.py changed, reloading.
Watching for file changes with StatReloader
Not Found: /favicon.ico
Watching for file changes with StatReloader
Internal Server Error: /
Traceback (most recent call last):
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 505, in parse
    compile_func = self.tags[command]
                   ~~~~~~~~~^^^^^^^^^
KeyError: 'static'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\projects\views.py", line 8, in home
    return render(request, 'projects/home.html', {'profile': profile, 'projects': projects, 'cv': cv})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\shortcuts.py", line 24, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\engine.py", line 175, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\engine.py", line 157, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 200, in compile_nodelist
    return parser.parse()
           ^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 513, in parse
    raise self.error(token, e)
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 511, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loader_tags.py", line 293, in do_extends
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 513, in parse
    raise self.error(token, e)
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 511, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loader_tags.py", line 232, in do_block
    nodelist = parser.parse(("endblock",))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 507, in parse
    self.invalid_block_tag(token, command, parse_until)
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 558, in invalid_block_tag
    raise self.error(
django.template.exceptions.TemplateSyntaxError: Invalid block tag on line 54: 'static', expected 'endblock'. Did you forget to register or load this tag?
Not Found: /favicon.ico
Internal Server Error: /
Traceback (most recent call last):
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 505, in parse
    compile_func = self.tags[command]
                   ~~~~~~~~~^^^^^^^^^
KeyError: 'static'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\projects\views.py", line 8, in home
    return render(request, 'projects/home.html', {'profile': profile, 'projects': projects, 'cv': cv})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\shortcuts.py", line 24, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\engine.py", line 175, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\engine.py", line 157, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 200, in compile_nodelist
    return parser.parse()
           ^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 513, in parse
    raise self.error(token, e)
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 511, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loader_tags.py", line 293, in do_extends
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 513, in parse
    raise self.error(token, e)
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 511, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loader_tags.py", line 232, in do_block
    nodelist = parser.parse(("endblock",))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 507, in parse
    self.invalid_block_tag(token, command, parse_until)
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 558, in invalid_block_tag
    raise self.error(
django.template.exceptions.TemplateSyntaxError: Invalid block tag on line 54: 'static', expected 'endblock'. Did you forget to register or load this tag?
Internal Server Error: /
Traceback (most recent call last):
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\projects\views.py", line 8, in home
    return render(request, 'projects/home.html', {'profile': profile, 'projects': projects, 'cv': cv})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\shortcuts.py", line 24, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\engine.py", line 175, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\engine.py", line 157, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 200, in compile_nodelist
    return parser.parse()
           ^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 513, in parse
    raise self.error(token, e)
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 511, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loader_tags.py", line 293, in do_extends
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 513, in parse
    raise self.error(token, e)
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 511, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\loader_tags.py", line 232, in do_block
    nodelist = parser.parse(("endblock",))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 513, in parse
    raise self.error(token, e)
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 511, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\defaulttags.py", line 853, in do_for
    nodelist_loop = parser.parse(
                    ^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 484, in parse
    raise self.error(token, e)
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 482, in parse
    filter_expression = self.compile_filter(token.contents)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 600, in compile_filter
    return FilterExpression(token, self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 698, in __init__
    filter_func = parser.find_filter(filter_name)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vmanb\OneDrive\Desktop\Games\CodeInsitute\VitorBarbosa\myenv\Lib\site-packages\django\template\base.py", line 606, in find_filter
    raise TemplateSyntaxError("Invalid filter: '%s'" % filter_name)
django.template.exceptions.TemplateSyntaxError: Invalid filter: 'truncate_words'
Watching for file changes with StatReloader
Watching for file changes with StatReloader
